edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java

```diff
@@ -1160,8 +1162,21 @@ public List getViewableStudentsForUser(Long gradebookId, String userId, List stu
 		List groupIds = new ArrayList(sectionIdStudentIdsMap.keySet());
 		List permsForGroupsAnyCategory = getPermissionsForUserForGroup(gradebookId, userId, groupIds);
 		
-		if (permsForGroupsAnyCategory.isEmpty()) {
-			return viewableStudents;
+		//FIXME: based on our realms checks earlier, sections should already be filtered to one that the
+		//User has access to. May need to make sure they have TA role in those sections
```

Comment:
It is not a good idea to assume that the sections are filtered to one the user all ready has access to, especially because this is a public method.